ext/drreg/drreg.c

```diff
@@ -169,6 +200,18 @@ get_where_app_pc(instr_t *where)
 }
 #endif
 
+/* Checks whether or not the passed register is in the appropriate range. */
+static inline bool
+is_applicable_simd(reg_id_t simd_reg)
```

Comment:
Doesn't this always return reg_is_vector_simd(simd_reg)?

Reply:
As explained [here](https://github.com/DynamoRIO/dynamorio/pull/3893#discussion_r335154973), this consider the range of SIMD vectors according to the architecture (statically determined). Shall I mark as resolved?

Reply:
So is is_applicable_simd() basically to check whether it is within 0 <= reg <= proc_num_simd_registers()?

Reply:
I instead use the static variant, namely MCXT_NUM_SIMD_SLOTS

Reply:
Resolved ?

Reply:
Use the enum for static structures that are not dynamically sizes, and proc_num_simd_registers() otherwise. proc_num_simd_registers() takes the OS and processor into account, mainly the distinction is there for AVX-512.

Reply:
i.e. I think instead of ranging it with DR_REG_APPLICABLE_*_SIMD, it should range it with DR_REG_START_ZMM - proc_num_simd_registers()

Reply:
Okay, will work on it soon.

Reply:
Obsolete, resolving.